tests/cross/src/test/scala/tests/pc/InlayHintsSuite.scala

```diff
@@ -1116,6 +1014,217 @@ class InlayHintsSuite extends BaseInlayHintsSuite {
+    """|import java.nio.file.Paths
+       |object Vars {
+       |  val a = Paths.get(".")
+       |}
+       |""".stripMargin,
+    """|import java.nio.file.Paths
+       |object Vars {
+       |  val a/*: Path<<java/nio/file/Path#>>*/ = Paths.get(".")
+       |}
+       |""".stripMargin
+  )
+  check(
+    "defaults",
+    """|import java.nio.file.Paths
+       |import java.nio.file.Path
+       |object Main {
+       |  case class Dependency()
+       |  object Dependency{
+       |    def of(
+       |      organization: String,
+       |      name: String,
+       |      version: String,
+       |    ): Dependency = ???
+       |  }
+       |  def downloadDependency(
+       |      dep: Dependency,
+       |      scalaVersion: Option[String] = None,
+       |      classfiers: Seq[String] = Seq.empty,
+       |      resolution: Option[Object] = None,
+       |  ): List[Path] = ???
+       |
+       |  def downloadSemanticdbJavac: List[Path] = {
+       |    downloadDependency(
+       |      Dependency.of(
+       |        "com.sourcegraph",
+       |        "semanticdb-javac",
+       |        "BuildInfo.javaSemanticdbVersion",
+       |      ),
+       |      None,
+       |    )
+       |  }
+       |
+       |  def downloadSemanticdbJavac2: List[Path] = {
+       |    downloadDependency(
+       |      Dependency.of(
+       |        "com.sourcegraph",
+       |        "semanticdb-javac",
+       |        "BuildInfo.javaSemanticdbVersion",
+       |      ),
+       |      None,
+       |      resolution = None
+       |    )
+       |  }
+       |
+       |}
+       |""".stripMargin,
+    """|package defaults
+       |import java.nio.file.Paths
+       |import java.nio.file.Path
+       |object Main {
+       |  case class Dependency()
+       |  object Dependency{
+       |    def of(
+       |      organization: String,
+       |      name: String,
+       |      version: String,
+       |    ): Dependency = ???
+       |  }
+       |  def downloadDependency(
+       |      dep: Dependency,
+       |      scalaVersion: Option[String] = None,
+       |      classfiers: Seq[String] = Seq.empty/*[Nothing<<scala/Nothing#>>]*/,
+       |      resolution: Option[Object] = None,
+       |  ): List[Path] = ???
+       |
+       |  def downloadSemanticdbJavac: List[Path] = {
+       |    downloadDependency(
+       |      /*dep = */Dependency.of(
+       |        /*organization = */"com.sourcegraph",
+       |        /*name = */"semanticdb-javac",
+       |        /*version = */"BuildInfo.javaSemanticdbVersion",
+       |      ),
+       |      /*scalaVersion = */None,
+       |    )
+       |  }
+       |
+       |  def downloadSemanticdbJavac2: List[Path] = {
+       |    downloadDependency(
+       |      Dependency.of(
+       |        /*organization = */"com.sourcegraph",
+       |        /*name = */"semanticdb-javac",
+       |        /*version = */"BuildInfo.javaSemanticdbVersion",
+       |      ),
+       |      None,
```

Comment:
This becomes much more complex now since all the parameters are now synthetic if we reorder the params in any way (though the flag is not set in reality). I might work on it later, but I will just do a follow up issue before mergeing. TODO